Formula/scotch.rb

```diff
@@ -55,15 +57,18 @@ def install
         return 0;
       }
     EOS
-    system ENV.cc, "test.c", "-L#{lib}", "-lscotch"
+    system ENV.cc, "test.c", "-L#{lib}", "-lscotch", "-lscotcherr",
+                             "-pthread", "-L#{Formula["zlib"].opt_lib}", "-lz", "-lm"
     assert_match version.to_s, shell_output("./a.out")
 
     system ENV.cc, pkgshare/"test_strat_seq.c", "-o", "test_strat_seq",
-           "-I#{include}", "-L#{lib}", "-lscotch", "-lscotcherr", "-lm", "-pthread"
+                   "-I#{include}", "-L#{lib}", "-lscotch", "-lscotcherr", "-lm", "-pthread",
+                   "-L#{Formula["zlib"].opt_lib}", "-lz"
     assert_match "Sequential mapping strategy, SCOTCH_STRATDEFAULT", shell_output("./test_strat_seq")
 
     system "mpicc", pkgshare/"test_strat_par.c", "-o", "test_strat_par",
-           "-I#{include}", "-L#{lib}", "-lptscotch", "-lscotch", "-lptscotcherr", "-lm", "-pthread"
+                    "-I#{include}", "-L#{lib}", "-lptscotch", "-lscotch", "-lptscotcherr", "-lm", "-pthread",
```

Comment:
For this test to work on Linux, I think `-Wl,-rpath,#{lib}` is also needed because `mpicc` may not use the GCC spec that automatically adds `-Wl,-rpath,HOMEBREW_PREFIX/lib`.

Reply:
Thanks for the hint, @danielnachun